File: deleteUsers/handler.py
import json
import os

import boto3

client = boto3.resource("dynamodb")

IS_OFFLINE = os.getenv("IS_OFFLINE", False)
if IS_OFFLINE:
    boto3.Session(
        aws_access_key_id="<aws_access_key_id>",
        aws_secret_access_key="<aws_secret_access_key>",
    )
    client = boto3.resource("dynamodb", endpoint_url="http://localhost:8000")

table = client.Table("usersTable")


def deleteUsers(event, context):
    user_id = event["pathParameters"]["id"]
    result = table.delete_item(Key={"pk": user_id})

    body = json.dumps({"message": f"user {user_id} deleted"})

    response = {
        "statusCode": result["ResponseMetadata"]["HTTPStatusCode"],
        "body": body,
    }

    return response

File: getusers/handler.js
const aws = require("aws-sdk")

let dynamoDBClientParams = {}

if (process.env.IS_OFFLINE) {
    dynamoDBClientParams =  {
        region: 'localhost',
        endpoint: 'http://localhost:8000',
        accessKeyId: '<aws_access_key_id>',  // needed if you don't have aws credentials at all in env
        secretAccessKey: '<aws_secret_access_key>' // needed if you don't have aws credentials at all in env
    }
}

const dynamodb = new aws.DynamoDB.DocumentClient(dynamoDBClientParams)

const getUsers = async (event, context) => {

    let userId = event.pathParameters.id

    var params = {
        ExpressionAttributeValues: { ':pk': userId },
        KeyConditionExpression: 'pk = :pk',
        TableName: 'usersTable'
    };

    return dynamodb.query(params).promise().then(res => {
        console.log(res)
        return {
            "statusCode": 200,
            "body": JSON.stringify({ 'user': res})
        }
    })
}

module.exports = {
    getUsers
}
